File: cpp-exercises/01_hello_internship.cpp
#include <iostream>
#include <string>

std::string build_welcome_message(const std::string& name) {
    if (name.empty()) {
        return "Hello, Student! Welcome to the internship.";
    }
 
    return "Hello, " + name + "! Welcome to the internship.";
}

int main() {
    std::string name;
    std::string topic;
    int age;
    std::cout << "Internship welcome program\n";
    std::cout << "Enter your name: ";
    std::getline(std::cin, name);

    std::cout << "How old are you: ";
    std::cin >> age;

    std::cout << "What do you want to practice most? ";
    std::getline(std::cin, topic);

    if (topic.empty()) {
        topic = "C++ and Python";
    }

    std::cout << '\n';
    std::cout << build_welcome_message(name) << '\n';
    std::cout << "Your focus for the internship is: " << topic << ".\n";
    std::cout << "Write small changes, test often, and ask clear questions.\n";

    return 0;
}

File: python-exercises/01_hello_internship.py
def build_welcome_message(name: str) -> str:
    clean_name = name.strip().title() or "Student"
    return f"Hello, {clean_name}! Welcome to the internship."


def main() -> None:
    print("Internship welcome program")
    name = input("Enter your name: ").strip()
    color = input("What is your favorite color: ").strip()
    favorite_topic = input("What do you want to practice most? ").strip()

    if not favorite_topic:
        favorite_topic = "Python and C++"

    print()
    print(build_welcome_message(name))
    print(f"Your focus for the internship is: {favorite_topic}.")
    print(f"You have a really good choice of color, {color} is a really nice color.")
    print("Small steps, clean commits, and clear questions matter.")
    print("You got it I believe in you.")
    print("Good luck!")


if __name__ == "__main__":
    main()
